frontend/src/metabase/palette/hooks/useCommandPalette.tsx

```diff
@@ -0,0 +1,187 @@
+import { Icon } from "metabase/ui";
+import type { SearchResult } from "metabase-types/api";
+
+export type PalettePageId = "root" | "admin_settings";
+
+export const useCommandPalette = ({
+  query,
+  debouncedSearchText,
+}: {
+  query: string;
+  debouncedSearchText: string;
+}) => {
+  const dispatch = useDispatch();
+
+  const hasQuery = query.length > 0;
+
+  const {
+    data: searchResults,
+    error: searchError,
+    isLoading: isSearchLoading,
+  } = useSearchListQuery<SearchResult>({
+    enabled: !!debouncedSearchText,
+    query: { q: debouncedSearchText, limit: 5 },
+    reload: true,
+  });
+
+  const { data: recentItems } = useRecentItemListQuery({
+    enabled: !debouncedSearchText,
+    reload: true,
+  });
+
+  const adminPaths = useSelector(getAdminPaths);
+  const settingValues = useSelector(getSettings);
+  const settingsSections = useMemo<Record<string, any>>(
+    () => getSectionsWithPlugins(),
+    [],
+  );
+
+  const docsAction = useMemo<KBarAction[]>(() => {
+    const ret: KBarAction[] = [
+      {
+        id: "search_docs",
+        name: query
+          ? `Search documentation for "${query}"`
+          : t`View documentation`,
+        section: "docs",
+        keywords: query, // Always match the query string
+        icon: () => <Icon name="document" />,
+        perform: () => {
+          const host = "https://www.metabase.com";
+          if (query) {
+            const params = new URLSearchParams({ query: query });
+            // TODO: find the documentation search URL in the right way
+            window.open(`${host}/search?${params}`);
+          } else {
+            window.open(`${host}/docs/latest`);
+          }
+        },
+      },
+    ];
+    return ret;
+  }, [query]);
+
+  useRegisterActions(hasQuery ? docsAction : [], [docsAction, hasQuery]);
+
+  const searchResultActions = useMemo<KBarAction[]>(() => {
+    const ret: KBarAction[] = [];
+    if (isSearchLoading) {
+      ret.push({
+        id: "search-is-loading",
+        name: "Loading...",
+        keywords: query,
+        section: "search",
+      });
+    } else if (searchError) {
+      ret.push({
+        id: "search-error",
+        name: t`Could not load search results`,
+        section: "search",
+      });
+    } else if (debouncedSearchText) {
+      if (searchResults?.length) {
+        ret.push(
+          ...searchResults.map(result => {
+            const wrappedResult = Search.wrapEntity(result, dispatch);
+            return {
+              id: `search-result-${result.id}`,
+              name: result.name,
+              icon: <Icon {...wrappedResult.getIcon()} />,
+              section: "search",
+              perform: () => {
+                dispatch(closeModal());
+                dispatch(push(wrappedResult.getUrl()));
+              },
+            };
+          }),
+        );
+      } else {
+        ret.push({
+          id: "no-search-results",
```

Comment:
this probably shouldn't be a result item, right?

If i select it, it kinda kills the palette.
![Screen Shot 2024-03-18 at 4 44 00 PM](https://github.com/metabase/metabase/assets/30528226/bf481eee-897c-400c-9fb6-a55a67780d7d)